src/naive/internals.rs

```diff
@@ -370,13 +387,17 @@ impl fmt::Debug for Of {
 /// The whole bits except for the least 3 bits are referred as `Mdl`
 /// (month, day of month and leap flag),
 /// which is an index to the `MDL_TO_OL` lookup table.
+///
+/// The methods implemented on `Mdf` do not always return a valid value.
+/// Dates than can't exist, like February 30, can still be represented.
```

Comment:
"than" -> "that"

Why do we need to allow `Mdf` instances with an invalid value?

Reply:
Testing whether an ordinal is valid is easy. Testing a month, day, leap year flag combination is valid, and doing it fast, is more involved.

The table lookup to convert `Mdf` to `Of` is pretty much the ideal place to put the validity check.